### PR TITLE
master

### PR DESCRIPTION
**Pull request by gayratv**
add cloneReadableStream.ts with new ES6 typescript syntax for Node 14 and 15

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "semi": true,
+  "trailingComma": "all",
+  "singleQuote": true,
+  "printWidth": 120,
+  "tabWidth": 2
+}
```
Comment (levansuper) [inline .prettierrc:7]: can we use  "useTabs": true instead of spaces please

**Review by levansuper — COMMENTED** [review]
First of all thanks for the PR, I really appreciate it.

Could you briefly explain the changes and the problems they solve?

File: test.cloneReadableStream.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+const readClone1 = new ReadableStreamClone(readStream, { highWaterMark: 3 });
+const readClone2 = new ReadableStreamClone(readStream, { highWaterMark: 3 });
+
+const writeStream1 = fs.createWriteStream('text1.txt');
+const writeStream2 = fs.createWriteStream('text2.txt');
+
+readClone1.pipe(writeStream1);
+readClone2.pipe(writeStream2);
+
+console.log('File test.txt was copied - OK.');
```
Comment (levansuper) [inline test.cloneReadableStream.ts]: I would like to stick to dash-case while naming the files and name the test files as clone-readable-stream.spec.ts
Also could we have a separate test folder for all the test files?

File: cloneReadableStream.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    readableStream.on('error', (err) => {
+      this.emit('error', err);
+    });
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  _read() {}
+}
+
+module.exports = ReadableStreamClone;
```
Comment (levansuper) [inline cloneReadableStream.ts]: dash-case for file names please

File: .prettierrc (hunk 1 above)
Comment (levansuper) [inline .prettierrc:7]: can we use "useTabs": true instead of spaces, please

**Comment by gayratv** [comment]
> First of all thanks for the PR, I really appreciate it.
> 
> Could you briefly explain the changes and the problems they solve?



> First of all thanks for the PR, I really appreciate it.
> 
> Could you briefly explain the changes and the problems they solve?

First of all, I would like to express my gratitude to you for your wonderful package.

This is my first PR - so I apologize for any possible mistakes :)

Your package works great on Node 14LTS, so there are no problems to solve.

But: the documentation says:
https://nodejs.org/dist/latest-v14.x/docs/api/util.html#util_util_inherits_constructor_superconstructor

"Usage of util.inherits () is discouraged. Please use the ES6 class and extends keywords to get language level inheritance support."

Actually, only for this reason, I decided to use ES6 syntax and rewrite the code a little.

File: .prettierrc (hunk 1 above)
Comment (gayratv) [reply .prettierrc:7]: I'm push new commit, please - see :)

**Comment by levansuper** [comment]
Closing this due to #6.
Maybe we could work on #10 instead

**Pull request closed without merging**